### PR TITLE
Ignore removing switch for mellanox platform due to known limitation

### PR DESCRIPTION
**Pull request by Junchao-Mellanox**
Why I did this?

Due to a limitation in mellanox platform, sonic ignores syncs graceful shutdown in syncd.sh previously. However, ignoring graceful shutdown in syncd.sh also ignores closing other resources such as removing all counters. 
This PR is going to ignore remove_switch for mellanox platform when shutdown type is cold. It make sure other shutdown operations are properly executed in shutdown flow.

How I test this?

Running config reload on many platforms and check result.

### CODE REVIEW

File: syncd/Syncd.cpp [hunk 1]
```diff
@@ -4738,7 +4739,11 @@ void Syncd::run()
 
     m_mdioIpcServer->stopMdioThread();
 
-    sai_status_t status = removeAllSwitches();
+    sai_status_t status = SAI_STATUS_SUCCESS;
+    if (shutdownType != SYNCD_RESTART_TYPE_COLD || 0 != strcmp(SONIC_ASIC_PLATFORM, "mellanox"))
```
Comment (nazariig) [inline syncd/Syncd.cpp]: @Junchao-Mellanox having a static string comparison looks very strange. Can we use a preprocessor to add a vendor specific logic instead? Also, one of the options is to export an environment variable like we do have in `swss`

**Review by nazariig — CHANGES_REQUESTED** [review]

File: configure.ac [hunk 2]
```diff
@@ -62,9 +67,9 @@ AC_ARG_ENABLE(syncd,
 esac],[syncd=true])
 AS_IF([test "x${ENABLESYNCD}" = "xno" ],[syncd=false],[])
 AM_CONDITIONAL(SYNCD, test x$syncd = xtrue)
-AM_COND_IF([SYNCD], [
-        AM_COND_IF([SAIVS], [], [AC_CHECK_LIB([sai], [main], [AC_MSG_NOTICE(libsai found)], [AC_MSG_ERROR(libsai is required for syncd)])
-])])
+#AM_COND_IF([SYNCD], [
+#        AM_COND_IF([SAIVS], [], [AC_CHECK_LIB([sai], [main], [AC_MSG_NOTICE(libsai found)], [AC_MSG_ERROR(libsai is required for syncd)])
+#])])
```
Comment (nazariig) [inline configure.ac]: @Junchao-Mellanox why this was removed?

**Review by nazariig — CHANGES_REQUESTED** [review]

File: configure.ac [hunk 3]
```diff
@@ -15,6 +15,11 @@ AM_PATH_PYTHON3
 AX_CODE_COVERAGE
 AX_ADD_AM_MACRO_STATIC([])
 
+# Generate a macro "#define MELLANOX_PLATFORM 1" in config.h
+if test x$CONFIGURED_PLATFORM = xmellanox; then
+    AC_DEFINE([MELLANOX], [1], [Define to 1 on Mellanox platform])
+fi
```
Comment (nazariig) [inline configure.ac]: @Junchao-Mellanox suggest to have:
```
AM_CONDITIONAL(SONIC_ASIC_PLATFORM_MELLANOX, test x$CONFIGURED_PLATFORM = xmellanox)
AM_COND_IF([SONIC_ASIC_PLATFORM_MELLANOX],
    [AC_DEFINE([MELLANOX_PLATFORM], [1], [Define to 1 on Mellanox platform])])
```

File: configure.ac (hunk 2 above)
Comment (Junchao-Mellanox) [reply configure.ac]: will fix

Comment (Junchao-Mellanox) [reply configure.ac]: fixed

File: syncd/Syncd.cpp [hunk 4]
```diff
@@ -1,3 +1,4 @@
+#include "config.h"
```
Comment (nazariig) [inline syncd/Syncd.cpp:1]: @Junchao-Mellanox why do we need this?

Comment (Junchao-Mellanox) [reply syncd/Syncd.cpp:1]: The macro `#define MELLNOX 1` is generated to config.h.

**Review by nazariig — APPROVED** [review]

**Pull request closed without merging**